Document new ingest and search pipeline allowlist settings (#7414)
opensearch-project/documentation-website: _ingest-pipelines/processors/index-processors.md

## Batch-enabled processors

Some processors support batch ingestion---they can process multiple documents at the same time as a batch. These batch-enabled processors usually provide better performance when using batch processing. For batch processing, use the [Bulk API]({{site.url}}{{site.baseurl}}/api-reference/document-apis/bulk/) and provide a `batch_size` parameter. All batch-enabled processors have a batch mode and a single-document mode. When you ingest documents using the `PUT` method, the processor functions in single-document mode and processes documents in series. Currently, only the `text_embedding` and `sparse_encoding` processors are batch enabled. All other processors process documents one at a time.

## Selectively enabling processors

> Processors defined by the [ingest-common module](https://github.com/opensearch-project/OpenSearch/blob/2.x/modules/ingest-common/src/main/java/org/opensearch/ingest/common/IngestCommonPlugin.java) can be selectively enabled by providing the `ingest-common.processors.allowed` cluster setting. If not provided, all processors are enabled by default. Specifying an empty list disables all processors. If the setting is changed to remove previously enabled processors, any pipeline using a disabled processor will fail after node restart when the new setting takes effect.

Reviewer: ```suggestion
Processors defined by the [ingest-common module](https://github.com/opensearch-project/OpenSearch/blob/2.x/modules/ingest-common/src/main/java/org/opensearch/ingest/common/IngestCommonPlugin.java) can be selectively enabled by providing the `ingest-common.processors.allowed` cluster setting. If not provided, then all processors are enabled by default. Specifying an empty list disables all processors. If the setting is changed to remove previously enabled processors, then any pipeline using a disabled processor will fail after node restart when the new setting takes effect.
```

Reviewer 2: @natebower One day "If, then" will stick for me 💯

> Processors defined by the [ingest-common module](https://github.com/opensearch-project/OpenSearch/blob/2.x/modules/ingest-common/src/main/java/org/opensearch/ingest/common/IngestCommonPlugin.java) can be selectively enabled by providing the static cluster setting `ingest-common.processors.allowed`. If not specified, then all processors are enabled by default. If an empty list is specified, then all processors are disabled. If the setting is changed to remove previously-enabled processors, then any pipeline using a disabled processor will fail after the node is restarted and the new setting takes effect.

Reviewer 2: ```suggestion
Processors defined by the [ingest-common module](https://github.com/opensearch-project/OpenSearch/blob/2.x/modules/ingest-common/src/main/java/org/opensearch/ingest/common/IngestCommonPlugin.java) can be selectively enabled by providing the static cluster setting `ingest-common.processors.allowed`. If not specified, then all processors are enabled by default. If an empty list is specified, then all processors are disabled. If the setting is changed to remove previously enabled processors, then any pipeline using a disabled processor will fail after the node is restarted and the new setting takes effect.
```

> Processors defined by the [ingest-common module](https://github.com/opensearch-project/OpenSearch/blob/2.x/modules/ingest-common/src/main/java/org/opensearch/ingest/common/IngestCommonPlugin.java) can be selectively enabled by providing the static cluster setting `ingest-common.processors.allowed`. If not specified, then all processors are enabled by default. If an empty list is specified, then all processors are disabled. If the setting is changed to remove previously enabled processors, then any pipeline using a disabled processor will fail after the node is restarted and the new setting takes effect.

Reviewer 2: ```suggestion
Processors defined by the [ingest-common module](https://github.com/opensearch-project/OpenSearch/blob/2.x/modules/ingest-common/src/main/java/org/opensearch/ingest/common/IngestCommonPlugin.java) can be enabled by providing the ingest-common.processors.allowed cluster setting. If not provided, all processors are enabled by default. Specifying an empty list disables all processors. If the setting is changed to remove previously enabled processors, any pipeline using a disabled processor will fail after node restart when the new setting takes effect.
```
